torch_np/random.py

```diff
@@ -0,0 +1,120 @@
+"""Wrapper to mimic (parts of) np.random API surface.
+
+NumPy has strict guarantees on reproducibility etc; here we don't give any.
+
+Q: default dtype is float64 in numpy
+
+"""
+from math import sqrt
+
+import torch
+
+from . import asarray
+from ._detail._scalar_types import default_float_type as _default_float_type
+from ._detail import _util
+
+_default_dtype = _default_float_type.torch_dtype
+
+__all__ = ["seed", "random_sample", "sample", "random", "rand", "randn", "normal"]
+
+
+def seed(seed=None):
+    if seed is not None:
+        torch.random.manual_seed()
+
+
+def random_sample(size=None):
+    if size is None:
+        values = torch.rand(())
+        return float(values)
+    else:
+        values = torch.rand(size).to(_default_dtype)
+        return asarray(values)
+
+
+def rand(*size):
+    return random_sample(size)
+
+
+sample = random_sample
+random = random_sample
+
+
+def uniform(low=0.0, high=1.0, size=None):
+    if size is None:
+        values = torch.rand(())
+        return float(low + (high - low) * values)
+    else:
+        values = torch.rand(size).to(_default_dtype)
+        return asarray(low + (high - low) * values)
+
+
+def randn(*size):
+    if size == ():
+        return float(torch.randn(size))
+    else:
+        values = torch.randn(*size).to(_default_dtypes)
+        return asarray(values)
+
+
+def normal(loc=0.0, scale=1.0, size=None):
+    if size is None:
+        size = ()
+    return loc + scale * randn(*size).to(_default_dtype)
```

Comment:
```suggestion
    return loc + scale * randn(*size)
```

Reply:
Why? Note that numpy's default is float64 not float32, and ISTM it's better to centralize this.

Reply:
This logic of setting the default type and handling the size is already implemented in the `randn` function, so no need to repeat it here.

Reply:
Ah, OK.  This is different in the last commit now, which implements 
`torch.empty(size, dtype=_default_dtype).uniform_(low, high)`, as requested.